addons/account/models/account_move_line.py

```diff
@@ -2211,7 +2211,10 @@ def _check_amls_exigibility_for_reconciliation(self, shadowed_aml_values=None):
             return
 
         if any(aml.reconciled for aml in self):
-            raise UserError(_("You are trying to reconcile some entries that are already reconciled."))
+            if all(aml.matching_number for aml in self):
+                raise UserError(_("You are trying to reconcile entries that are all already reconciled."))
+            if not all(aml.matching_number.startswith('P') for aml in self if aml.matching_number):
+                raise UserError(_("You can only reconcile already reconciled entries if they are partially so."))
```

Comment:
1) You can't change anything in such complex code without writing a test
2) If `reconciled` is set, it means both `amount_residual` & `amount_residual_currency` are 0 so "You are trying to reconcile some entries that are already reconciled.": I don't understand your use case then